coderd/httpmw/workspaceagent.go

```diff
@@ -18,40 +18,67 @@ import (
 
 type workspaceAgentContextKey struct{}
 
+func WorkspaceAgentOptional(r *http.Request) (database.WorkspaceAgent, bool) {
+	user, ok := r.Context().Value(workspaceAgentContextKey{}).(database.WorkspaceAgent)
+	return user, ok
+}
+
 // WorkspaceAgent returns the workspace agent from the ExtractAgent handler.
 func WorkspaceAgent(r *http.Request) database.WorkspaceAgent {
-	user, ok := r.Context().Value(workspaceAgentContextKey{}).(database.WorkspaceAgent)
+	user, ok := WorkspaceAgentOptional(r)
 	if !ok {
 		panic("developer error: agent middleware not provided")
```

Comment:
This panic could now be confusing, since it's potentially provided but made optional.